app/src/main/java/org/kiwix/kiwixmobile/localfiletransfer/LocalFileTransferActivity.kt

```diff
@@ -0,0 +1,391 @@
+ * app. Two devices are connected to each other using WiFi Direct, followed by file transfer.
+ *
+ * File transfer involves two phases:
+ * 1) Handshake with the selected peer device, using [PeerGroupHandshakeAsyncTask]
+ * 2) After handshake, starting the files transfer using [SenderDeviceAsyncTask] on the sender
+ * device and [ReceiverDeviceAsyncTask] files receiving device
+ */
+@SuppressLint("GoogleAppIndexingApiWarning", "Registered")
+class LocalFileTransferActivity : BaseActivity(),
+  WifiDirectManager.Callbacks {
+  @Inject
+  lateinit var alertDialogShower: AlertDialogShower
+
+  @Inject
+  lateinit var wifiDirectManager: WifiDirectManager
+
+  @Inject
+  lateinit var locationManager: LocationManager
+
+  private var isFileSender = false // Whether the device is the file sender or not
+  private var filesForTransfer = ArrayList<FileItem>()
+  private var fileListAdapter: FileListAdapter? = null
+  private var wifiPeerListAdapter: WifiPeerListAdapter? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_local_file_transfer)
+
+    /*
+     * Presence of file Uris decides whether the device with the activity open is a sender or receiver:
+     * - On the sender device, this activity is started from the app chooser post selection
+     * of files to share in the Library
+     * - On the receiver device, the activity is started directly from within the 'Get Content'
+     * activity, without any file Uris
+     * */
+    val filesIntent = intent
+    val fileUriArrayList: ArrayList<Uri>?
+    fileUriArrayList = filesIntent.getParcelableArrayListExtra(Intent.EXTRA_STREAM)
+    isFileSender = fileUriArrayList?.isNotEmpty() == true && fileUriArrayList.size > 0
+
+    val toolbar: Toolbar =
+      findViewById(R.id.toolbar)
+    // I know this is not the best method, but android extensions were showing 'unknown reference' error
+    setSupportActionBar(toolbar)
+    toolbar.setNavigationIcon(R.drawable.ic_close_white_24dp)
+    toolbar.setNavigationOnClickListener { finish() }
+    wifiPeerListAdapter = WifiPeerListAdapter(
+      WifiP2pDelegate(wifiDirectManager::sendToDevice)
+    )
+
+    list_peer_devices?.adapter = wifiPeerListAdapter
+    list_peer_devices?.layoutManager = LinearLayoutManager(this)
+    list_peer_devices?.setHasFixedSize(true)
+    if (isFileSender) {
+      fileUriArrayList.map {
+        filesForTransfer.add(FileItem(it))
+      }
+      displayFileTransferProgress(filesForTransfer)
+    }
+    wifiDirectManager.startWifiDirectManager(filesForTransfer)
+  }
+
+  override fun onCreateOptionsMenu(menu: Menu): Boolean {
+    menuInflater.inflate(R.menu.wifi_file_share_items, menu)
+    return true
+  }
+
+  override fun onOptionsItemSelected(item: MenuItem): Boolean {
+    var status = false
+    return if (item.itemId == R.id.menu_item_search_devices) {
```

Comment:
![image](https://user-images.githubusercontent.com/3358282/89772955-dd6feb00-dafa-11ea-8a11-7e2609f5c9de.png)
This flow has been radically altered, such a change should not occur in a conversion ticket